lib/rubocop/cop/rspec/return_from_stub.rb

```diff
@@ -84,6 +92,99 @@ def check_block_body(node)
         def dynamic?(node)
           !node.recursive_literal?
         end
+
+        # :nodoc:
+        class AndReturnCallCorrector
+          def initialize(node, config)
+            @node = node
+            @receiver, _method_name, @args = *node
+            @config = config
+          end
+
+          def call(corrector)
+            # Heredoc autocorrection is not yet implemented.
+            return if heredoc?
+
+            if args.single_line?
+              corrector.replace(range, " { #{args.source} }")
+            else
+              corrector.replace(range, multiline_replacement)
+            end
+          end
+
+          private
+
+          attr_reader :node, :receiver, :args
+          attr_reader :config
+
+          def heredoc?
+            args.loc.is_a?(Parser::Source::Map::Heredoc)
+          end
+
+          def range
+            Parser::Source::Range.new(
+              node.source_range.source_buffer,
+              receiver.source_range.end_pos,
+              node.source_range.end_pos
+            )
+          end
+
+          def multiline_replacement
+            buffer = StringIO.new
+            buffer << " do\n"
+            args.source.lines.each do |line|
+              buffer << "#{indentation}#{line.strip}\n"
```

Comment:
How much indentation logic does this cop need to know? I assume that the Layout cops will be run again after each autocorrect anyway, and fix any (most of the) wrong indentation.

Reply:
IMO we should not go there. Even if we replace with `{ }` type of block, it's another cop that will say (and auto-correct) multi-line blocks to do/end.

I'll test on a real code-base tomorrow and let you know how it works.